src/lib.rs

```diff
@@ -1,25 +1,165 @@
+//!     // In GDAL you can read arbitrary regions of the raster, and have them up- or down-sampled
+//!     // when the output buffer size is different from the read size. The terminology GDAL
+//!     // uses takes getting used to. All parameters here are in pixel coordinates.
+//!     // Also note, tuples are in `(x, y)`/`(cols, rows)` order.
+//!     // `window` is the (x, y) coordinate of the upper left corner of the region to read.
+//!     let window = (20, 30);
+//!     // `window_size` is the amount to read `(cols, rows)`
+//!     let window_size = (2, 3);
+//!     // `size` is the output buffer size. If this is different from `window_size`, then
+//!     // the `resample_alg` parameter below becomes relevant.
+//!     let size = (2, 3);
+//!     // Options here include `NearestNeighbor` (default), `Bilinear`, `Cubic`, etc.
+//!     let resample_alg = None;
+//!     // Note the `u8` type parameter. GDAL will convert the native pixel type to whatever is
+//!     // specified here... which may or may not be right for your use case!
+//!     let rv = band.read_as::<u8>(window, window_size, size, resample_alg)?;
+//!     // `Rasterband::read_as` returns a `Buffer` struct, which contains the shape of the output
+//!     // `(cols, rows)` and a `Vec<_>` containing the pixel values.
+//!     println!("    Data size: {:?}", rv.size);
+//!     println!("    Data values: {:?}", rv.data);
+//! }
+//! # Ok(())
+//! # }
+//! ```
+//! The resulting output is:
+//! ```text
+//! Dataset description: fixtures/tinymarble.tif
+//! Number of bands: 3
+//!   Band 1
+//!     Description: ''
+//!     No-data value: None
+//!     Pixel data type: 1
+//!     Scale: None
+//!     Offset: None
+//!     Data size: (2, 3)
+//!     Data values: [47, 74, 77, 118, 98, 122]
+//!   Band 2
+//!     Description: ''
+//!     No-data value: None
+//!     Pixel data type: 1
+//!     Scale: None
+//!     Offset: None
+//!     Data size: (2, 3)
+//!     Data values: [50, 79, 77, 118, 95, 119]
+//!   Band 3
+//!     Description: ''
+//!     No-data value: None
+//!     Pixel data type: 1
+//!     Scale: None
+//!     Offset: None
+//!     Data size: (2, 3)
+//!     Data values: [71, 94, 79, 115, 77, 98]
 //! ```
-//! use std::path::Path;
-//! use gdal::Dataset;
-//! use gdal::vector::LayerAccess;
 //!
-//! let dataset = Dataset::open(Path::new("fixtures/roads.geojson")).unwrap();
-//! let mut layer = dataset.layer(0).unwrap();
-//! for feature in layer.features() {
-//!     let highway_field = feature.field("highway").unwrap().unwrap();
-//!     let geometry = feature.geometry();
-//!     println!("{} {}", highway_field.into_string().unwrap(), geometry.wkt().unwrap());
+//! ### Vector
+//!
+//! > _Note_: Describing the sophisticated structure of GDAL vector data is outside the scope of this example, so
+//! > please see [Vector Data Model](https://gdal.org/user/vector_data_model.html) section of the GDAL user manual for details.
+//!
+//! ```rust
```

Comment:
Maybe add a couple of newlines to spread this out a little and make it nicer to read?